torch/nn/modules/batchnorm.py

```diff
@@ -145,13 +145,15 @@ class _LazyBatchNorm(LazyModuleMixin, _BatchNorm):
 
     def __init__(self, eps=1e-5, momentum=0.1, affine=True, track_running_stats=True):
         super(_LazyBatchNorm, self).__init__(
-            0, eps, momentum, affine, track_running_stats)
+            0, eps, momentum, False, False)
+        self.affine = affine
+        self.track_running_stats = track_running_stats
         if self.affine:
             self.weight = UninitializedParameter()
             self.bias = UninitializedParameter()
         if self.track_running_stats:
-            self.running_mean = UninitializedBuffer()
-            self.running_var = UninitializedBuffer()
+            self.register_buffer('running_mean', UninitializedBuffer())
```

Comment:
@albanD This will probably be solved after #35735 is done. And if there isn't anyone who is working on it, I'd love to give it a try~

Reply:
Well... I just checked the logic of `register_buffer` and `__setattr__` again and I'm really sorry that I was wrong with this change. The `register_buffer` can only be called once and the subsequent assignment will change the corresponding value. I have changed it back to where is was.

Reply:
> Also, I found that the buffers in _NormBase was set to parameter when track_running_stats is disabled.

Wow, most likely not no!
A quick blame shows that this has been here since this was first introduced so possibly a typo at that time.
cc @jbschlosser can you think of any good reason for us to do that beyond a typo?

Reply:
Given that `running_mean` / `running_var` are basically the poster children for the use of buffers, I can't think of any reason for them to specifically be parameters! Perhaps `None` wasn't supported for `register_buffer` back then?